src/toxcore/messenger/packet/messenger/mod.rs

```diff
@@ -0,0 +1,161 @@
+/*! Top-level Messenger Packets
+*/
+
+use crate::toxcore::binary_io::*;
+
+mod online;
+mod action;
+mod offline;
+mod message;
+mod nickname;
+mod msi;
+mod file_control;
+mod typing;
+mod user_status;
+mod file_data;
+mod status_message;
+mod file_send_request;
+
+pub use self::online::*;
+pub use self::action::*;
+pub use self::offline::*;
+pub use self::message::*;
+pub use self::nickname::*;
+pub use self::msi::*;
+pub use self::file_control::*;
+pub use self::typing::*;
+pub use self::user_status::*;
+pub use self::file_data::*;
+pub use self::status_message::*;
+pub use self::file_send_request::*;
+
+/** Messenger packet enum that encapsulates all types of Messenger packets.
+*/
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum Packet {
```

Comment:
this should be in `messenger/packet/mod.rs`

Reply:
I changed structure of directory because of this problem https://github.com/tox-rs/tox/pull/358#discussion_r279135935